Assets/HoloToolkit/Utilities/Editor/SceneLauncherEditor.cs

```diff
@@ -0,0 +1,150 @@
+            EditorGUILayout.IntSlider(buttonRowMaxProperty, 1, 10);
+            serializedObject.ApplyModifiedProperties();
+        }
+
+        private void ShowList(SerializedProperty list)
+        {
+            bool updateBuildScenes = false;
+
+            // property name with expansion widget
+            EditorGUILayout.PropertyField(list);
+
+            if (list.arraySize == 0)
+            {
+                int index = list.arraySize;
+                list.InsertArrayElementAtIndex(index);
+                list.GetArrayElementAtIndex(index).stringValue = SceneManager.GetActiveScene().path;
+                updateBuildScenes = true;
+            }
+
+            if (list.isExpanded)
+            {
+                EditorGUI.indentLevel++;
+
+                // add button row
+                EditorGUILayout.BeginHorizontal();
+                GUILayout.FlexibleSpace();
+                // the add element button
+                if (GUILayout.Button(addButtonContent, EditorStyles.miniButton, miniButtonWidth))
+                {
+                    var index = list.arraySize;
+                    list.InsertArrayElementAtIndex(index);
+                    list.GetArrayElementAtIndex(index).stringValue = string.Empty;
+                }
+
+                EditorGUILayout.EndHorizontal();
+
+                // Scene rows
+                for (int index = 0; index < list.arraySize; index++)
+                {
+                    string path = list.GetArrayElementAtIndex(index).stringValue;
+                    var sceneAsset = AssetDatabase.LoadAssetAtPath(path, typeof(SceneAsset)) as SceneAsset;
+
+                    EditorGUILayout.BeginHorizontal();
+                    GUI.enabled = index != sceneLauncher.SceneLauncherBuildIndex;
+
+                    // the element
+                    var sceneObj = EditorGUILayout.ObjectField(sceneAsset, typeof(SceneAsset), false);
+
+                    if (GUI.changed)
+                    {
+                        path = sceneObj != null ? AssetDatabase.GetAssetPath(sceneObj) : string.Empty;
+                        updateBuildScenes = true;
+                    }
+
+                    list.GetArrayElementAtIndex(index).stringValue = path;
+
+                    // the remove element button
+                    if (GUILayout.Button(removeButtonContent, EditorStyles.miniButton, miniButtonWidth))
+                    {
+                        list.DeleteArrayElementAtIndex(index);
+                        updateBuildScenes = true;
+                    }
+
+                    GUI.enabled = true;
+                    EditorGUILayout.EndHorizontal();
+                }
+
+                EditorGUI.indentLevel--;
+            }
+
+            if (updateBuildScenes && list.arraySize > 0)
+            {
+                ListBuildScenes(list);
+            }
+        }
+
+        private void CheckBuildScenes(SerializedProperty list)
```

Comment:
> - A scene is removed from the build settings and a different one is added. (Lists will have the same length, but different elements.)

In my latest update this is no longer possible.

> - Build settings are updated without opening the SceneLauncher prefab in the inspector, so update doesn't occur.

Do you mean that it isn't serialized into the prefab?  Then I believe you are correct this does not handle that edge case.

> - Build settings are updated while SceneLauncher prefab is already open in inspector, so OnEnable isn't called again.

Fixed.  Updates the prefab list when the user navigates back into the editor from the build window.

> - SceneLauncher prefab is modified by this function, but Unity is closed without saving the change to the prefab. (Would editor code be required to mark it dirty or save the prefab?)

I think making sure people intentionally save changes is better.  We could also prompt to ask if people want to save before quitting.